client/client.go

```diff
@@ -94,10 +97,18 @@ func (client *CqlClient) String() string {
 // CqlClientConnection.InitiateHandshake. Alternatively, use ConnectAndInit to get a fully-initialized connection.
 func (client *CqlClient) Connect(ctx context.Context) (*CqlClientConnection, error) {
 	log.Debug().Msgf("%v: connecting", client)
-	dialer := net.Dialer{}
+	var conn net.Conn
+	var err error
 	connectCtx, connectCancel := context.WithTimeout(ctx, client.ConnectTimeout)
 	defer connectCancel()
-	if conn, err := dialer.DialContext(connectCtx, "tcp", client.RemoteAddress); err != nil {
+	if client.TLSConfig != nil {
```

Comment:
I'm not sure one needs the if-else branches, just `dialer := tls.Dialer{Config: client.TLSConfig}` should work in all cases, wdyt?

Reply:
The dialers themselves are different. One is `net.Dialer` and the ther is `tls.Dialer`. Are you suggesting that we could use `tls.Dialer` for non TLS connections? Never tried that tbh

Reply:
Oh sorry my bad, I didn't notice the difference. Please disregard.